pkg/manifest/metadata.go

```diff
@@ -112,6 +112,34 @@ func (m Metadata) EffectiveReadingProgression() ReadingProgression {
 	return LTR
 }
 
+// InferredAccessibility returns the accessibility metadata inferred from the
+// manifest and stored in OtherMetadata.
+func (m Metadata) InferredAccessibility() *A11y {
+	var a11y *A11y
+	if a11yJSON, ok := m.OtherMetadata["inferredAccessibility"].(map[string]interface{}); ok {
+		a11y, _ = A11yFromJSON(a11yJSON)
+	}
+	return a11y
+}
+
+// SetOtherMetadata marshalls the value to a JSON map before storing it in
+// OtherMetadata under the given key.
+func (m Metadata) SetOtherMetadata(key string, value interface{}) error {
+	bytes, err := json.Marshal(value)
+	if err != nil {
+		return err
+	}
+	var object map[string]interface{}
+	err = json.Unmarshal(bytes, &object)
+	if err != nil {
+		return err
+	}
+
+	m.OtherMetadata[key] = object
+
+	return nil
+}
```

Comment:
Can you explain why you're marhsalling and then unmarshalling the passed-in value? Is this in order to do a deep copy?

Reply:
`OtherMetadata` must contain "raw" JSON objects (as in `map[string]interface{}`) and not complex types. Because we want the same results as if this was parsed from a `manifest.json` file.

I didn't find a way to go directly from `struct` -> `map[string]interface{}`, hence the weird dance with `Marshall`/`Unmarshall`.

Reply:
@mickael-menu how about this? https://github.com/mitchellh/mapstructure . If you want I can build a performance benchmark to compare, but my gut instinct tells me this will definitely be more performant than the JSON way.

Reply:
Wouldn't this bypass any custom `Marshall` implementation in our structs?

Reply:
Which API would you use for this? I see only decoding APIs, but here we want to encode a native struct to `map[string]interface{}` using the JSON rules of each struct.

Reply:
Yeah, on second thought you're right... However, I have a new idea: We add an interface like this:
```go
interface Mappable {
    ToMap() (map[string]interface{}, error)
}
```
And the function first attempts to load the supplied `value` as something that implements that interface and call that function. If that's not possible, then it falls back to your json approach.

Reply:
Sounds good 👍

Reply:
I named it `JSONMappable` to make it explicit that the objects in the map must be JSON-compatible.